### DISCUSSION TITLE
Local dev environment

### DISCUSSION THREAD

**Discussion by mbainter** [mondoohq/cnquery#1732] [Q&A] [answered]
It looks like in the main branch we're restructuring towards having providers as plugins. It's the goal to eventually split those into their own repositories similar to the hashicorp provider split?

Also, can someone throw together some unofficial notes on how to work with the system in its current alpha state in a local build environment?

I've been trying to test a recent change I made and evaluate some additional possibilities, but I can't seem to get cnquery to accept the local build of the AWS provider. 

I realize we're only just now hitting an alpha release so updating the contribution docs is probably premature. But some informal updates here around how the workflow has changed would be helpful.

**Comment by vjeffrey** [comment, marked as answer]
hi @mbainter, thank you for your contribution! 
for now/the near future all the providers will remain in this repository.
we'll get some dev env instructions up on how to work with things in the current state, thanks for the suggestion!
we'll get working on that this morning

**Reply by vjeffrey** [reply]
provider debug instructions: https://github.com/mondoohq/cnquery/pull/1733
